trulens_eval/trulens_eval/feedback/provider/hugs.py

```diff
@@ -612,10 +651,86 @@ def hallucination_evaluator(
             raise RuntimeError(
                 "Unexpected response type. Please check the API endpoint."
             )
-
         return score
 
 
+class HuggingfaceLocal(HuggingfaceBase):
+    """
+    Out of the box feedback functions calling Huggingface APIs.
+    """
+    _cached_tokenizers: Dict[str, PreTrainedTokenizerBase] = {}
+    _cached_models: Dict[str, PreTrainedModel] = {}
+
+    def _retrieve_tokenizer_and_model(
+        self, key: str
+    ) -> Tuple[PreTrainedTokenizerBase, PreTrainedModel]:
+        if key not in self._cached_tokenizers:
+            self._cached_tokenizers[key] = AutoTokenizer.from_pretrained(key)
+        if key not in self._cached_models:
+            self._cached_models[
+                key] = AutoModelForSequenceClassification.from_pretrained(key)
+        tokenizer = self._cached_tokenizers[key]
+        model = self._cached_models[key]
+        return tokenizer, model
+
+    def _language_scores_endpoint(self, text: str) -> Dict[str, float]:
+        raise NotImplementedError(
+            "Currently not implemented in for local Huggingface!"
+        )
+
+    def _context_relevance_endpoint(self, input: str) -> float:
+        raise NotImplementedError(
+            "Currently not implemented in for local Huggingface!"
+        )
+
+    def _positive_sentiment_endpoint(self, input: str) -> float:
+        raise NotImplementedError(
+            "Currently not implemented in for local Huggingface!"
+        )
+
+    def _toxic_endpoint(self, input: str) -> float:
+        raise NotImplementedError(
+            "Currently not implemented in for local Huggingface!"
+        )
+
+    def _summarized_groundedness_endpoint(self, input: str) -> float:
+        raise NotImplementedError(
+            "Currently not implemented in for local Huggingface!"
+        )
+
+    # TODEP
+    @_tci
+    def _doc_groundedness(self, premise: str, hypothesis: str) -> float:
+        import torch
+        tokenizer, model = self._retrieve_tokenizer_and_model(
+            HUGS_DOCNLI_MODEL_PATH
```

Comment:
can we allow users to choose the path?

Reply:
This corresponds to the same model that the inference API is hard-coded too.